packages/shared/client/ReplayClient.ts

```diff
@@ -878,6 +887,36 @@ export class ReplayClient implements ReplayClientInterface {
     }
   }
 
+  async waitForLoadedRegions(focusRange: TimeStampedPointRange | PointRange | null): Promise<void> {
+    return new Promise(resolve => {
+      const checkLoaded = () => {
+        const loadedRegions = this.loadedRegions;
+        let isLoaded = false;
+        if (loadedRegions !== null) {
+          if (focusRange !== null) {
+            const pointRange = toPointRange(focusRange);
+
+            isLoaded = isRangeInRegions(pointRange.begin, pointRange.end, loadedRegions.loaded);
+          } else {
+            isLoaded = loadedRegions.loaded.length > 0 && loadedRegions.loading.length === 0;
```

Comment:
Oh. This isn't right. `loading` will never get cleared. `loaded` will just equal `loading` 🤪